### PR TITLE
ENG-254: Include default .editorconfig file

### PR DESCRIPTION
**Pull request by stefaniuk**
Fixes nhs-england-tools/repository-template#3

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+root = true
```
Comment (amaanibn-nasar1-nhs) [inline .editorconfig:1]: I've not come across this kind of file before
Can I assume that it works similar to something like a `.prettierrc` file?
Is this editor independent or does it only work for vscode?

File: README.md [hunk 2]
```diff
@@ -1,60 +1,60 @@
...
-## Contacts
-
-Provide a way to contact the owners of this project. It can be a team, an individual or information on the means of getting in touch via active communication channels, e.g. opening a GitHub discussion, raising an issue, etc.
-
-## Licence
-
-> The [LICENCE.md](./LICENCE.md) file will need to be updated with the correct year and owner
-
-A copyright notice should be included here as well.
+# Repository Template
```
Comment (amaanibn-nasar1-nhs) [inline README.md:1]: I can't seem to see any changes in this file.
so I'd assume the change is due to the above file and it's added `\n` to the end of each line or something similar?

Comment (stefaniuk) [reply README.md:1]: It converted platform-specific line ending from `\r` to `\n` which is enforced by the `.editorconfig` rule.

File: .editorconfig (hunk 1 above)
Comment (stefaniuk) [reply .editorconfig:1]: That's correct, `root = true` is used to tell the rule processor that this `.editorconfig` file is the one and the only in this repository to stop it from traversing the whole directory structure.

Comment (stefaniuk) [reply .editorconfig:1]: There is a VS Code plugin for it https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig. EditorConfig is supported by all major editors and IDEs.

**Review by stefaniuk — COMMENTED** [review]
LGTM as this is a standard .editorconfig file content

File: README.md [hunk 3]
```diff
@@ -1,60 +1,60 @@
...
+
+## Usage
+
+After a successful installation, provide an informative example of how this project can be used. Additional code snippets, screenshots and demos work well in this space. You may also link to the other documentation resources, e.g. the [User Guide](./docs/user-guide.md) to demonstrate more use cases and to show more features.
+
+## Architecture
+
+### Diagrams
+
+The [C4 model](https://c4model.com/) is a simple and intuitive way to create software architecture diagrams that are clear, consistent, scalable and most importantly collaborative. This should result in documenting all the system interfaces, external dependencies and integration points.
```
[the remark this answers is not in the document]
Comment (stefaniuk) [reply README.md:36]: Thanks, @jope1. I added a task to the backlog to provide guidance on that.

File: README.md [hunk 4]
```diff
@@ -1,60 +1,60 @@
...
+
+### Configuration
+
+Most of the projects are built with customisability and extendability in mind. At a minimum, this can be achieved by implementing a service level configuration options and settings. The intention of this section is to show how this can be used. If the system processes data, you could mention here for example how the input is prepared for testing - anonymised, synthetic or live data.
```
[the remark this answers is not in the document]
Comment (stefaniuk) [reply README.md:40]: This has been corrected. Thank you!

**Pull request merged**